src/utils/generateEventLink.ts

```diff
@@ -0,0 +1,77 @@
+import { CalendarEvent as CalendarLinkEvent, google, ics } from 'calendar-link';
+
+export const enum CalendarType {
+    Google,
+    ICS,
+}
+
+export interface CalendarEventDate {
+    // from 1 to 31
+    day: number;
+    // from 0 to 11
+    month: number;
+    year: number;
+    time?: {
+        // from 0 to 23
+        hours: number;
+        // from 0 to 59
+        minutes: number;
+    };
+}
+
+export interface CalendarEvent {
+    title: string;
+    allDay: boolean;
+    location?: string;
+    start: CalendarEventDate;
+    end: CalendarEventDate;
+}
+
+/**
+ * Generates calendar link for an event by a calendar type
+ * @param event Event
+ * @param calendarType CalendarType
+ */
+export function generateEventLink(
+    event: CalendarEvent,
+    calendarType: CalendarType,
+): string {
+    const calendarEvent = generateCalendarEvent(event);
+    switch (calendarType) {
+        case CalendarType.Google:
+            return google(calendarEvent);
+        case CalendarType.ICS:
+            return decodeURIComponent(
+                ics(calendarEvent).split('charset=utf8,')[1],
```

Comment:
Maybe take it out and put into a constant with a meaningful name to avoid "magic strings"? E.g. `ICS_CALENDAR_EVENT_DEVIDER`